tests/regression.rs

```diff
@@ -0,0 +1,380 @@
+//! Regression tests for _most_ types that implement `serde::Serialize`.
+//!
+//! For remaining types see: ./regression_opcodes.rs
+//!
+//! If you find a type defined in `rust-bitcoin` that implements `Serialize` and does _not_ have a
+//! regression test please add it.
+//!
+//! Types/tests were found using, and are ordered by, the output of: `git grep -l Serialize`.
+//!
+
+#![allow(unused_imports)] // All imports are unused when use-serde feature is not set.
+
+extern crate bitcoin;
+extern crate bincode;
+extern crate secp256k1;
+
+use std::collections::BTreeMap;
+use std::fs::File;
+use std::io::{self, Read, BufReader};
+use std::io::BufRead;
+use std::str::FromStr;
+
+use bincode::serialize;
+
+use bitcoin::{Address, Network, Script, Block, TxIn, TxOut, Transaction, PrivateKey, PublicKey, SchnorrSig, SchnorrSigHashType, EcdsaSigHashType, EcdsaSig, Txid, OutPoint};
+use bitcoin::blockdata::witness::Witness;
+use bitcoin::util::bip32::{ExtendedPrivKey, ExtendedPubKey, ChildNumber, KeySource};
+use bitcoin::util::uint::{Uint128, Uint256};
+use bitcoin::util::taproot::{ControlBlock, TaprootBuilder, LeafVersion};
+use bitcoin::util::psbt::{Input, Output, PartiallySignedTransaction};
+use bitcoin::util::psbt::raw::{self, Pair, Key, ProprietaryKey};
+use bitcoin::util::ecdsa;
+use bitcoin::hashes::hex::FromHex;
+use bitcoin::hashes::{sha256d, Hash, ripemd160, hash160, sha256};
+use bitcoin::consensus::encode::deserialize;
+
+/// Reads a comma separated list of integers from file `name` into a vector. Integers represent the
+/// expected bincode for a serialized data type.
+#[cfg(feature = "serde")]
+fn bincode_from_file(name: &str) -> Vec<u8> {
+    let name = format!("tests/data/bincode/{}", name);
+    let file = File::open(name).unwrap();
+    let reader = BufReader::new(file);
+
+    reader.lines().next().unwrap().unwrap()
+        .split(',').map(|s| s.trim())
+        .filter(|s| !s.is_empty())
+        .map(|s| s.parse().unwrap())
+        .collect()
+}
```

Comment:
Maybe store it as raw bytes and use `include_bytes!()` instead?

Anyway, since the file contains a single line using `std::fs::read_to_string()` would be simpler and faster.

Reply:
Used both these ideas as suggested :)